stdlib/http/post.go

```diff
@@ -16,111 +11,118 @@ import (
-				header.Object().Range(func(k string, v values.Value) {
-					if v.Type() == semantic.String {
-						req.Header.Set(k, v.Str())
-					} else {
-						rangeErr = errors.Newf(codes.Invalid, "header value %q must be a string", k)
-					}
-				})
-				if rangeErr != nil {
-					return nil, rangeErr
+		// Add headers to request
+		header, ok := args.Get("headers")
+		if ok && !header.IsNull() {
+			var rangeErr error
+			header.Object().Range(func(k string, v values.Value) {
+				if v.Type() == semantic.String {
+					req.Header.Set(k, v.Str())
+				} else {
+					rangeErr = errors.Newf(codes.Invalid, "header value %q must be a string", k)
 				}
+			})
+			if rangeErr != nil {
+				return nil, rangeErr
 			}
+		}
+
+		// Perform request
+		dc, err := deps.HTTPClient()
+		if err != nil {
+			return nil, errors.Wrap(err, codes.Aborted, "missing client in http.post")
+		}
 
-			// Perform request
-			dc, err := deps.HTTPClient()
+		statusCode, err := func(req *http.Request) (int, error) {
+			s, cctx := opentracing.StartSpanFromContext(ctx, "http.post")
+			s.SetTag("url", req.URL.String())
+			defer s.Finish()
+
+			req = req.WithContext(cctx)
+			response, err := dc.Do(req)
 			if err != nil {
-				return nil, errors.Wrap(err, codes.Aborted, "missing client in http.post")
+				return 0, err
 			}
 
-			statusCode, err := func(req *http.Request) (int, error) {
-				s, cctx := opentracing.StartSpanFromContext(ctx, "http.post")
-				s.SetTag("url", req.URL.String())
-				defer s.Finish()
+			// Read the response body but limit how much we will read.
+			// This is to allow a socket to be reused after it is closed.
+			wc := iocounter.Writer{Writer: ioutil.Discard}
+			r := io.LimitedReader{
+				R: response.Body,
+				N: maxResponseBody,
+			}
+			_, _ = io.Copy(&wc, &r)
+			_ = response.Body.Close()
+			s.LogFields(
+				log.Int("statusCode", response.StatusCode),
+				log.Int64("responseSize", wc.Count()),
+			)
+			return response.StatusCode, nil
+		}(req)
+		if err != nil {
+			return nil, err
+		}
 
-				req = req.WithContext(cctx)
-				response, err := dc.Do(req)
-				if err != nil {
-					return 0, err
-				}
+		// return status code
+		return values.NewInt(int64(statusCode)), nil
+	},
+	true, // post has side-effects
+)
 
-				// Read the response body but limit how much we will read.
-				// This is to allow a socket to be reused after it is closed.
-				wc := iocounter.Writer{Writer: ioutil.Discard}
-				r := io.LimitedReader{
-					R: response.Body,
-					N: maxResponseBody,
-				}
-				_, _ = io.Copy(&wc, &r)
-				_ = response.Body.Close()
-				s.LogFields(
-					log.Int("statusCode", response.StatusCode),
-					log.Int64("responseSize", wc.Count()),
-				)
-				return response.StatusCode, nil
-			}(req)
-			if err != nil {
-				return nil, err
-			}
+func init() {
+	flux.RegisterPackageValue("http", "post", post)
 
-			// return status code
-			return values.NewInt(int64(statusCode)), nil
-		},
-		true, // post has side-effects
-	))
```

Comment:
can you revert the changes to this file as they are not related to this PR anymore?

Reply:
I'm sure there's an easier way to revert then copy/paste from master, but I've reverted to the original and tested again.